app/controllers/api_controller.rb

```diff
@@ -0,0 +1,28 @@
+class ApiController < ApplicationController
+  def service_interaction_for_local_authority
+    @authority = LocalAuthority.find_by_slug!(params[:authority_slug])
+    @service = Service.find_by(lgsl_code: params[:lgsl])
+    interactions = @service.interactions
+    @link = links.detect { |l| interactions.first.service_interactions.first.id == l.service_interaction.id }
```

Comment:
Actually, I think this code wouldn't work properly at the moment as we're not targeting the right Interaction (defined by `param[:lgil]`). General concept still stands though :)